api/app/services/feedback_service.py

```
@@ -738,6 +745,344 @@ def get_source_weights(self) -> Dict[str, float]:
+            item
+            for item in feedback_items
+            if item.is_negative and (item.explanation or item.has_no_source_response)
+        ]
+
+        # Feedback by month
+        monthly_counts = defaultdict(int)
+        for item in feedback_items:
+            try:
+                month_key = item.timestamp[:7]  # YYYY-MM
+                monthly_counts[month_key] += 1
+            except:
```

Ruff (0.13.1)

Do not use bare `except`

(E722)